pkg/optimization/device_perf_helper.go

```diff
@@ -27,13 +27,14 @@ import (
 // Right now we are only supporing basic profile
 // Other advanced profiles to come later
 func IsValidPerfProfile(profile string) bool {
-	return strings.EqualFold(profile, consts.PerfProfileBasic) || strings.EqualFold(profile, consts.PerfProfileNone)
+	return isPerfTuningEnabled(profile) || strings.EqualFold(profile, consts.PerfProfileNone)
 }
 
 // getDiskPerfAttributes gets the per tuning mode and profile set in attributes
-func GetDiskPerfAttributes(attributes map[string]string) (profile, accountType, diskSizeGibStr, diskIopsStr, diskBwMbpsStr string, err error) {
+func GetDiskPerfAttributes(attributes map[string]string) (profile, accountType, diskSizeGibStr, diskIopsStr, diskBwMbpsStr string, deviceSettings map[string]string, err error) {
```

Comment:
Instead of returning a tuple of values, consider returning a struct containing the values. This will result is less churn for any future changes.

Reply:
Thanks for the feedback. Allow me to do this as a separate PR.